### PR TITLE
Transfer alertify.js to git auto-update, add v0.4.0rc1

### LINKED ISSUE #7519: [request] add fabien-d/alertify.js
https://github.com/fabien-d/alertify.js

### PR DESCRIPTION
**Pull request by sashberd**
close #7519, cc @fabien-d

Pull request for issue: #7519 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/fabien-d/alertify.js
 * Official website (optional, not the repository): http://fabien-d.github.com/alertify.js/
 * NPM package url (optional): https://www.npmjs.com/package/alertify
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 203
   - Count of stars: 4364
   - Count of forks: 794
 * NPM download stats (optional):
   - Downloads in the last day: 8
   - Downloads in the last week: 144
   - Downloads in the last month: 778

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [X] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [X] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/alertify/package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "alertify",
+  "description": "An unobtrusive customizable JavaScript notification system",
+  "version": "0.3.11",
+  "homepage": "https://fabien-d.github.com/alertify.js/",
```
Comment (maruilian11) [inline ajax/libs/alertify/package.json]: `https` seems to not work on this site, let's use `http`.

File: ajax/libs/alertify/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "alert",
+    "notification"
+  ],
+  "filename": "alertify.min.js",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/fabien-d/alertify.js.git",
+    "fileMap": [
+      {
+        "basePath": "lib",
```
Comment (maruilian11) [inline ajax/libs/alertify/package.json]: I think we also need another `basePath` for 0.4.0rc1. Thanks.

File: ajax/libs/alertify/package.json (hunk 1 above)
Comment (sashberd) [reply ajax/libs/alertify/package.json]: @maruilian11 It is working for me

File: ajax/libs/alertify/package.json (hunk 2 above)
Comment (sashberd) [reply ajax/libs/alertify/package.json]: Are we hosting alpha, beta and rc versions?

**Comment by sashberd** [comment]
@maruilian11 The PR was updated

File: ajax/libs/alertify/package.json (hunk 2 above)
Comment (maruilian11) [reply ajax/libs/alertify/package.json]: Yes.

**Comment by maruilian11** [comment]
Please mention the minify tool in commit msg. Thanks.

**Comment by sashberd** [comment]
@maruilian11 The PR was updated. FYR

File: ajax/libs/alertify/package.json (hunk 1 above)
Comment (extend1994) [reply ajax/libs/alertify/package.json]: Works for me too
@maruilian11 maybe try again?

**Comment by sashberd** [comment]
@extend1994  I suggest to update package.json to use git autoupdate in order to get those versions. Also, there are more releases on git instead of npm. I will update PR FYR

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

**Comment by sashberd** [comment]
@extend1994 ping

File: ajax/libs/alertify.js/package.json [hunk 3]
```diff
@@ -13,16 +13,30 @@
...
+        "basePath": "lib",
+        "files": [
+          "alertify*"
+        ]
+      },
+      {
+        "basePath": "themes",
+        "files": [
+          "alertify*"
+        ]
```
Comment (sufuf3) [inline ajax/libs/alertify.js/package.json]: Let's make the pattern as 
```js
        "basePath": "",
        "files": [
          "themes/alertify*"
        ]
```
We don't let `themes` as the basePath.
Also, if you saw v0.4.0rc1 which get the files from `dist`, the file structure is 
```sh
├── 0.4.0rc1
│   ├── alertify.js
│   ├── alertify.min.js
│   └── themes
│       ├── alertify.bootstrap.css
│       ├── alertify.css
│       └── alertify.default.css
```

Thanks.

**Comment by sashberd** [comment]
@sufuf3 v0.5.0 is from [npm](https://libraries.io/npm/alertify-js/0.5.0)

**Comment by sashberd** [comment]
@sufuf3 The PR message was updated. FYR

**Comment by sashberd** [comment]
@PeterDaveHello in rc of v0.4.0 it is in the `dist `folder... Should I remove it?

**Comment by sashberd** [comment]
@PeterDaveHello OK, I will remove `dist ` from from pattern. Should I open new issue to add version 0.4.0 that is rc?

**Comment by sashberd** [comment]
@sufuf3 @extend1994 The PR was updated. FYR

**Comment by PeterDaveHello** [comment]
@sashberd you can do it in the same PR.

**Comment by sashberd** [comment]
@PeterDaveHello The PR was updated. FYR

**Comment by sashberd** [comment]
@PeterDaveHello ping

**Comment by PeterDaveHello** [comment]
@sashberd I suppose you need review from @sufuf3 & @extend1994 first? I helped rebase the branch now.

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

**Comment by PeterDaveHello** [comment]
@sashberd looks like there is a commit from the bot should be removed?

**Comment by sashberd** [comment]
@PeterDaveHello The PR was fixed. FYR

File: ajax/libs/alertify.js/package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "filename": "alertify.min.js",
   "version": "0.5.0",
   "description": "Alertify is an unobtrusive customizable JavaScript notification system.",
-  "homepage": "http://fabien-d.github.io/alertify.js/",
+  "homepage": "https://fabien-d.github.io/alertify.js/",
```
Comment (extend1994) [inline ajax/libs/alertify.js/package.json]: Please make this change in another commit, thanks.

Comment (sashberd) [reply ajax/libs/alertify.js/package.json]: What do you mean in another commit? I made it in `Update alertify-js package.json to use git autoupdate  …`

Comment (extend1994) [reply ajax/libs/alertify.js/package.json]: Use new commit to do it.

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

**Comment by extend1994** [comment]
@PeterDaveHello Please help review this PR, thank you.

**Comment by sufuf3** [comment]
@sashberd Sorry that is it possible to mention lib's name in third commit message?
Because `Update homepage property in packge.json to SSL ` didn't show the lib's name, IMO, it's better show the library name in commit message that help people know the commit action.
What do you think?
Thanks

**Comment by sashberd** [comment]
@sufuf3 The name was mentioned in commit message

**Comment by sufuf3** [comment]
@PeterDaveHello Would you please review this PR?
Many thanks
